openfisca_france/model/prelevements_obligatoires/prelevements_sociaux/cotisations_sociales/allegements.py

```diff
@@ -415,6 +416,31 @@ def formula_2015_01_01(individu, period, parameters):
         return allegement * not_(stagiaire) * not_(apprenti) * non_cumulee
 
 
+class allegement_cotisation_maladie(Variable):
+    value_type = float
+    entity = Individu
+    definition_period = MONTH
+    label = "Allègement des cotisations employeur d’assurance maladie sur les bas et moyens salaires (Ex-CICE)"
+    reference = "https://www.legifrance.gouv.fr/codes/article_lc/LEGIARTI000037947559"
+
+    def formula_2019_01_01(individu, period, parameters):
+        assiette_allegement = individu('assiette_allegement', period)
+
+        # taux global tout salaire à réduire :
+        # openfisca_france/parameters/prelevements_sociaux/cotisations_securite_sociale_regime_general/mmid/taux.yaml
+        tranches_bareme_maladie = parameters(period).prelevements_sociaux.cotisations_securite_sociale_regime_general.mmid.taux
+        bareme_a_deux_tranches = (len(tranches_bareme_maladie.rates) == 2) * (len(tranches_bareme_maladie.thresholds) == 2)
+
+        taux_reduction = bareme_a_deux_tranches * (tranches_bareme_maladie.rates[1] - tranches_bareme_maladie.rates[0])
+        seuil_reduction = bareme_a_deux_tranches * tranches_bareme_maladie.thresholds[1]  # en nombre de smic
```

Comment:
Nous cherchons ici à calculer un allègement de cotisation qui est [décrit dans la loi](https://www.legifrance.gouv.fr/codes/article_lc/LEGIARTI000037947559) comme étant une réduction de **6 points** sur le taux normal de la cotisation (13% de taux normal). 

Or nous avons le paramètre [prelevements_sociaux.cotisations_securite_sociale_regime_general.mmid.taux](https://github.com/openfisca/openfisca-france/blob/66f9c83ac24349e0325f143e3779acd7352115fc/openfisca_france/parameters/prelevements_sociaux/cotisations_securite_sociale_regime_general/mmid/taux.yaml#L2-L14) sur le sujet donc il serait logique de l'utiliser. Mais : 
* le paramètre est un barème qui fait référence aux taux appliqués (taux normal de 13%, taux réduit de 7% = 13 - 6) et non pas au taux de l'allègement que nous voulons calculer (les fameux 6 points). 
* et c'est un barème harmonisé avec les [barèmes IPP](https://www.ipp.eu/baremes-ipp/prelevements-sociaux/1/cotisations_securite_sociale_ss_regime_general/mmid) (cf. taux réduit et normal au seuil de 2.5 SMIC).

D'où les questions suivantes @benjello @Sasha-Laniece : 
* Voyez-vous une meilleure façon de calculer le taux et le seuil (2.5 SMIC) de la réduction ci-dessus ? 
   > Il me semble que le code est plutôt moche et en particulier pour la vérification qu'il requière du nombre de tranches alors que cela est hors sujet du point de vue de la réduction : elle passerait par exemple à zéro si on ajoutait une tranche à l'avenir et il me semble qu'un `>= 2` ne garantirait guère plus qu'on calcule bien les 6 points au seuil de 2.5 SMIC 😢 
* Sommes-nous obligés de conserver ce barème ? Il me semble que oui 😕. 
  > En ne reprenant pas le chiffre `6` indiqué dans la loi, il me semble que ce barème comprend en lui-même un calcul mais que parce qu'il entre dans le champ des cotisations et du preprocessing, on doit le conserver pour le moment. D'autant qu'on ne veut pas ajouter de paramètre pour la réduction parce que cela ferait doublon, n'est-ce pas ? 
* Est-ce qu'on s'autorise à modifier le yaml harmonisé ? Ne serait-ce que pour la [référence législative](https://www.legifrance.gouv.fr/loda/article_lc/LEGIARTI000037949208/) qui ne cite pas les chiffres du barème alors que l'on a [cette référence de taux normal](https://www.legifrance.gouv.fr/codes/article_lc/LEGIARTI000036679615/) et [cette référence pour l'allègement et le seuil](https://www.legifrance.gouv.fr/codes/article_lc/LEGIARTI000037947559).

Reply:
ℹ️ En employant le taux par défaut de cotisation maladie (13%) pour déduire le taux d'allègement, on ne prend pas en considération les catégories spécifiques de salariés rattachés au régime général bénéficient d’un taux réduit dérogatoire de cotisation d’assurance maladie.
Le [boss](https://boss.gouv.fr/portail/accueil/allegements-generaux.html#titre_chapitre-2---la-reduction-du-tau-section-2---determination-du-mon-i-principes-generaux_b-reference-a-la-remuneration-an) donne cet exemple : 
```
Pour les artistes du spectacle et les mannequins, dont le taux de cotisation d’assurance maladie est fixé par dérogation à 9,1 % au lieu de 13% dans le droit commun, la réduction proportionnelle s'élèvera à 6 × (9,1 / 13) = 4,2 points. Le taux de la cotisation d’assurance maladie pour les rémunérations inférieures ou égales à 2,5 SMIC sera donc de 4,9 %.
```

Reply:
Je pense qu'il faut corriger le barème utilisé par openfsca pour utiliser les anciens taux et garder les paramètres de l'allègement à part. A nous de faire la modification sur les barèmes IPP en temps voulu. 

Pour les mannequins et les artistes du spectacles, on ne applique déjà pas le taux exact, donc l'allègement ...

cc @pzuldp

Reply:
d'accord avec @benjello et pour en principe répercuter ça côté barèmes IPP ; ceci nous "invite" effectivement à mieux réfléchir à l'articulation barèmes "légaux" et barèmes "effectifs" côté IPP, leur stockage, leur présentation, etc...
on ne code pas encore le régime dérogatoire, mais maintenant que ce sera facilité par une plus belle formule côté régime de base, ça motivera sans doute quelqu'un.e :)

Reply:
(Je me réveille tard mais:)
Je suis d'accord avec vous, c'est vraiment mieux que les 6 points soient un paramètre à part et non un calcul dans le barème.
Je trouve que c'est beaucoup plus lisible en termes de vérification (si on veut checker que l'allegement est bien codé), de représentation de la loi, et si jamais on veut amender ces 6 points

Reply:
Donc, si j'ai bien saisi : on remplace le barème actuel par un barème n'incluant pas la réduction et on crée un nouveau paramètre qui représente la réduction en elle-même. On obtiendrait alors deux paramètres openfisca-france qui correspondent directement à ce que dit la loi.

Reply:
Suite à nos échanges, un paramètre de taux d'allègement a été créé. 
Puis, on a identifié qu'il existait déjà un [répertoire](https://github.com/openfisca/openfisca-france/tree/df0eff95f9485a6bfca881e77e4c5458cf4cce75/openfisca_france/parameters/prelevements_sociaux/reductions_cotisations_sociales/alleg_gen) pour les allègements généraux de cotisations sociales. Ce paramètre y a donc été placé. 

Le seuil de 2.5 SMIC étant un élément de l'allègement, un paramètre le contenant a été créé dans le même répertoire.

Enfin, le taux normal a été transformé en paramètre simple. Ceci semble cohérent avec la documentation initialement présente dans ce paramètre harmonisé à savoir : `A partir de 2019, le taux réduit et le taux normal constitue bien des taux globaux d'imposition, et non des taux d'imposition par tranche, contrairement à précédemment` ([Src IPP](https://framagit.org/french-tax-and-benefit-tables/baremes-ipp-yaml/-/blob/29e3fee96e720a337cfc289f8a48a84c37c73e98/parameters/prelevements_sociaux/cotisations_securite_sociale_ss_regime_general/mmid.yaml)). Néanmoins, la référence législative évoque un taux normal et une réduction (il n'y aurait pas mention du taux réduit en tant que tel).